lib/developer_portal/lib/liquid/drops/message.rb

```diff
@@ -31,7 +31,7 @@ def created_at
       desc "URL of the message detail, points either to inbox or outbox."
       def url
         if @model.hidden_at.present?
-          admin_messages_trash_path(@model.id)
+          admin_messages_trash_path(id)
         else
           if @model.is_a?(MessageRecipient)
             admin_messages_inbox_path(@model.id)
```

Comment:
`@model.id` would actually return `@model.message.id` for the MessageRecipient :upside_down_face: 

I know it's confusing...

Reply:
Sorry sorry sorry, I got it all wrong and confused myself and you @jlledom :sweat_smile: 

I was referring to line `:34`, for the trash URL, we use the `Message` ID, and so we use `id`, which returns `@model.id` for `Message`, or `@model.message.id` for `MessageRecipient`.

Your comment however was about like `:37` - on forming the Inbox URL. So, in Inbox we're using `MessageRecipient`'s IDs... that's expected.

Which makes me think... maybe using Message's ID instead of MessageRecipient's could have some issues... :thinking: let me double-check.

Reply:
@jlledom I've checked this again. This is the same exact behavior as in the admin portal.
While the message is in the Inbox, it shows `MessageRecipient` with extra metadata (i.e. whether the message has been read or not), and the URLs are `MessageRecipient`'s. When the message is moved to Trash, the `Message` is shown - so in Trash we can't see whether the message was read or not, but once it's restored back to Inbox, we can see this again.

In the built-in forms in the dev portal we do not show **To:** field for received messages, so it doesn't matter whether we're rendering `Message` or `MessageRecipient` in  Trash, the subject, from address and the body are the same for both, as `MessageRecipient` delegates these fields to `Message`.